src/poprox_storage/repositories/data_stores/db.py

```diff
@@ -18,58 +17,29 @@
 logger.setLevel(logging.DEBUG)
 
 
-DbRepositories: type[BaseModel] | None = None
+def inject_repos(handler):
+    @wraps(handler)
+    def wrapper(event, context):
+        with DB_ENGINE.connect() as conn:
+            # this will be a dictionary containing all parameters (with type annotations)
+            params = get_type_hints(handler)
+            # remove event, context, and return type if they were annotated.
+            params.pop("event", None)
+            params.pop("context", None)
+            params.pop("return", None)
 
+            repos = dict()
+            for param, class_obj in params.items():
+                if class_obj in DatabaseRepository._repository_types:
+                    repos[param] = class_obj(conn)
 
-def db_repositories(*args):
-    create_repository_model()
+            return handler(event, context, **repos)
 
-    if len(args) == 1 and not isinstance(args[0], list):
-        names = [args[0]]
-    else:
-        names = args
-
-    def inner_decorator(handler):
-        @wraps(handler)
-        def wrapper(event, context):
-            type_hints = get_type_hints(DbRepositories)
-            with DB_ENGINE.connect() as conn:
-                # create Repositories object and pre-construct all requested repository objects
-                repos = DbRepositories(conn=conn)
-                for name in names:
-                    repo_type = type_hints[name]
-                    setattr(repos, name, repo_type(conn))
-                return handler(event, context, repos)
-
-        return wrapper
-
-    return inner_decorator
-
-
-def create_repository_model():
-    global DbRepositories
-
-    if DbRepositories is None:
-        # Convert the attribute names and classes to fields
-        fields = {
-            name: Annotated[type_, Field(default=None)] for name, type_ in DatabaseRepository._repository_types.items()
-        }
-
-        # Dynamically create a Pydantic model with those fields
-        DbRepositories = create_model(
-            "DbRepositories",
-            conn=(Connection, None),
-            **fields,
-            __config__=ConfigDict(arbitrary_types_allowed=True),
-        )
-
-
-def camel_to_snake(s):
-    return "".join(["_" + c.lower() if c.isupper() else c for c in s]).lstrip("_")
+    return wrapper
 
 
 class DatabaseRepository:
-    _repository_types = {}
+    _repository_types = set()
```

Comment:
Switched to set since we don't care what the parameter is named anymore -- this does formally require annotating parameters, but working with this type of magic that's probably for the best.